scylla/src/transport/session.rs

```diff
@@ -1192,7 +1192,12 @@ fn calculate_token(
 async fn resolve_hostname(hostname: &str) -> Result<SocketAddr, NewSessionError> {
     let failed_err = NewSessionError::FailedToResolveAddress(hostname.to_string());
     let mut ret = None;
-    for a in lookup_host(hostname).await? {
+    let addrs: Vec<SocketAddr> = match lookup_host(hostname).await {
+        Ok(addrs) => addrs.collect(),
+        // Use a default port in case of error, but propagate the original error on failure
+        Err(e) => lookup_host((hostname, 9042)).await.or(Err(e))?.collect(),
```

Comment:
The approach of appending a port at the end in case of error looks a bit hacky to me... But I don't have an idea for a better alternative :(

Reply:
Originally I thought of looking for `:` in the lookup string and decide from there, but string parsing also sucks, and here we at least have a neat tuple API for `lookup_host`

Reply:
I thought that looking for `:` would suffice, too, but apparently they are used for encoding IPv6 addresses: https://doc.rust-lang.org/std/net/struct.SocketAddrV6.html